src/main/java/org/minions/devfund/damianvp/battleship/Ocean.java

```diff
@@ -0,0 +1,189 @@
+package org.minions.devfund.damianvp.battleship;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Ocean class.
+ */
+public class Ocean {
+    static final int LENGTH = 20;
+    static final int TOTAL_SHIPS = 13;
+    private Ship[][] ships = new Ship[LENGTH][LENGTH];
+    private int shotsFired;
+    private int hitCount;
+    private int shipsSunk;
+    private Map<String, Integer> fleet;
+
+    /**
+     * Ocean constructor.
+     */
+    Ocean() {
+        for (int i = 0; i < LENGTH; i++) {
+            for (int j = 0; j < LENGTH; j++) {
+                EmptySea emptySea = new EmptySea();
+                this.ships[i][j] = emptySea;
+                emptySea.placeShipAt(i, j, true, this);
+            }
+        }
+        this.shotsFired = 0;
+        this.hitCount = 0;
+        this.shipsSunk = 0;
+        setFleet();
+    }
+
+    /**
+     * This method sets the quantity of ships allowed in the game.
+     */
+    private void setFleet() {
+        final int destroyer = 3;
+        final int submarine = 4;
+        fleet = new HashMap<>();
+        fleet.put("BattleShip", 1);
+        fleet.put("BattleCruiser", 1);
+        fleet.put("Cruiser", 2);
+        fleet.put("LightCruiser", 2);
+        fleet.put("Destroyer", destroyer);
+        fleet.put("Submarine", submarine);
```

Comment:
This should be initialized as static and the method should be removed.
```java
    private static final int BATTLESHIP_NUMBER = 1;
    private static final int BATTLE_CRUISER_NUMBER = 1;
    private static final int CRUISER_NUMBER = 2;
    private static final int LIGHT_CRUISER_NUMBER = 2;
    private static final int DESTROYER_NUMBER = 3;
    private static final int SUBMARINE_NUMBER = 4;
    private static final Map<String, Integer> FLEET = new HashMap<>();
    static {
        FLEET.put("BattleShip", BATTLESHIP_NUMBER);
        FLEET.put("BattleCruiser", BATTLE_CRUISER_NUMBER);
        FLEET.put("Cruiser", CRUISER_NUMBER);
        FLEET.put("LightCruiser", LIGHT_CRUISER_NUMBER);
        FLEET.put("Destroyer", DESTROYER_NUMBER);
        FLEET.put("Submarine", SUBMARINE_NUMBER);
    }
```

Reply:
@MinionsGroup/minions-h 
You can take as example this implementation.

Reply:
cool, I will apply this, thanks